torch/_inductor/dependencies.py

```diff
@@ -154,6 +154,13 @@ def merge(self, other: "ReadWrites"):
             op_counts = other.op_counts
         return ReadWrites(reads - writes, writes, index_exprs, op_counts=op_counts)
 
+    @staticmethod
+    def merge_list(read_writes: List["ReadWrites"]):
+        all_writes = set.union(*[rw.writes for rw in read_writes])
+        all_reads = set.union(*[rw.reads - all_writes for rw in read_writes])
```

Comment:
@mlazos It might be faster to do the `- all_writes` outside the union.  Worth benchmarking that.

Reply:
Doesn't look like it made a difference, I'll put outside since it simplifies the code.